### PR TITLE
[DOCS] Naming consistency of `length` functions

### PR DESCRIPTION
**Pull request by vicky1999**
Solves #2769 

- Added `length` function to Expression.list
- Added deprecation warning to `Expression.list.lengths`

### CODE REVIEW

File: docs/source/api_docs/expressions.rst [hunk 1]
```diff
@@ -215,6 +215,7 @@ List
 
    Expression.list.join
    Expression.list.lengths
```
[the remark this answers is not in the document]
Comment (vicky1999) [reply docs/source/api_docs/expressions.rst]: Done

**Comment by vicky1999** [comment]
@kevinzwang   Changes were made as per the comments.  Let me know if there are any other changes needed.

**Comment by vicky1999** [comment]
@kevinzwang 
 we also have `lengths` function in `series`.  So, created the `length` function and made `lengths` deprecated.

File: daft/expressions/expressions.py [hunk 2]
```diff
@@ -2936,6 +2937,19 @@ def count(self, mode: CountMode = CountMode.Valid) -> Expression:
     def lengths(self) -> Expression:
         """Gets the length of each list
 
+        Returns:
+            Expression: a UInt64 expression which is the length of each list
+        """
```
Comment (kevinzwang) [inline daft/expressions/expressions.py:2942]: Could you replace this docstring with something along the lines of 
> "(DEPRECATED) Please use `Expression.list.length` instead"

**Review by kevinzwang — APPROVED** [review]
Looks great! Let me know once you've resolved the one comment I had and I can merge this.

**Comment by vicky1999** [comment]
@kevinzwang Docstring of `lengths` function is updated as per your comments.

**Comment by kevinzwang** [comment]
@vicky1999 enabling auto-merge. Thank you for this contribution!

**Pull request merged**